flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/deduplicate/MiniBatchDeduplicateKeepFirstRowFunction.java

```diff
@@ -46,14 +47,18 @@
 	// state stores a boolean flag to indicate whether key appears before.
 	private ValueState<Boolean> state;
 
-	public MiniBatchDeduplicateKeepFirstRowFunction(TypeSerializer<BaseRow> typeSerializer) {
+	public MiniBatchDeduplicateKeepFirstRowFunction(TypeSerializer<BaseRow> typeSerializer, long minRetentionTime) {
+		super(minRetentionTime);
 		this.typeSerializer = typeSerializer;
 	}
 
 	@Override
 	public void open(ExecutionContext ctx) throws Exception {
 		super.open(ctx);
 		ValueStateDescriptor<Boolean> stateDesc = new ValueStateDescriptor<>("existsState", Types.BOOLEAN);
+		if (!stateTtlConfig.equals(StateTtlConfig.DISABLED)){
```

Comment:
use `stateTtlConfig.isEnabled`. We should also correct the implementation in `JoinRecordStateViews`.

Reply:
I see, tks

Reply:
By the way, should I correct the implementation in JoinRecordStateViews?

Reply:
Yes. Thanks.